storage/registers.go

```diff
@@ -6,27 +6,32 @@ import (
 
 // RegisterIndex defines methods for the register index.
 type RegisterIndex interface {
-	// Get register by the register ID at a given block height.
+	// Get the register by the ID at a given block height or lower.
 	//
-	// If the register at the given height was not indexed, returns the highest
-	// height the register was indexed at.
+	// If the register at the given height was not indexed, return the highest height the register was indexed at if one exists.
 	// Expected errors:
 	// - storage.ErrHeightNotIndexed if the given height was not indexed yet or lower than the first indexed height.
 	// - storage.ErrNotFound if the given height is indexed, but the register does not exist.
 	Get(ID flow.RegisterID, height uint64) (flow.RegisterValue, error)
 
-	// LatestHeight returns the latest indexed height.
+	// LatestHeight at which we indexed the registers. 
+	// The value will continue to change as we index new registers.
+	// Expected errors:
+	// - storage.ErrNotFound if the latest height is not set.
 	LatestHeight() (uint64, error)
 
-	// FirstHeight at which we started to index. Returns the first indexed height found in the store.
+	// FirstHeight at which we indexed the registers. 
+	// The value should not be changed after the register index is created.
+	// Expected errors:
+	// - storage.ErrNotFound if the first height is not set.
```

Comment:
I thought we decided to require that `FirstHeight` and `LatestHeight` were set before initializing the storage?

Reply:
We did. Wonder how you think we can improve this? if we have to require the first and last being set on initialization of the storage then I guess we need to split this into two interfaces one for importing and one for the current use case, otherwise if we dissalow initialization if values are not set then we wont be able to init for the case of importing. In any case I feel that might be refactored with the importing PR? but I'm open to change this now if you think it's better.